runtime/doc/treesitter.txt

```diff
@@ -481,6 +481,19 @@ library.
 ==============================================================================
 Lua module: vim.treesitter                               *lua-treesitter-core*
 
+foldexpr({lnum})                                   *vim.treesitter.foldexpr()*
+    Returns the fold level for {lnum} in the current buffer. Can be set
+    directly to 'foldexpr': >lua
+
+     vim.wo.foldexpr = 'v:lua.vim.treesitter.foldexpr()'
```

Comment:
Ideally `foldmethod=treesitter` or `foldmethod=syntax` would do this internally. But we haven't figured out the full story for bridging / integrating legacy "syntax" mechanisms with "treesitter".

Reply:
Yes, this is deliberately meant as an intermediary step before both the frontend (full `:syntax` integration) and the backend (extranges based folding) is in place. As it's a self-contained feature, it makes sense to already start getting feedback on the actual folding experience.

Reply:
This is only a stopgap solution, but at least allows us to get the queries in the codebase. A proper folding solution wouldn't use `foldexpr` at all and instead would use TBD API functions.